src/p2p/net_node.inl

```diff
@@ -2297,6 +2297,15 @@ namespace nodetool
     return true;
   }
   //-----------------------------------------------------------------------------------
+  epee::net_utils::network_address get_preferred_address(const epee::net_utils::connection_context_base& context, basic_node_data& node_data)
```

Comment:
One last note: 
Could the last parameter be made a `const` reference? So:
`epee::net_utils::network_address get_preferred_address(const epee::net_utils::connection_context_base& context, const basic_node_data& node_data)`
Because without it, it appears that this is an output parameter, while it's not. I'm assuming that the `get_type_id()` is a const method.

Reply:
Once this is done, then it's OK from my side.